bertopic/representation/_openai.py

```diff
@@ -217,11 +218,21 @@ def extract_topics(self,
                 else:
                     response = self.client.chat.completions.create(**kwargs)
 
-                # Check whether content was actually generated
-                # Adresses #1570 for potential issues with OpenAI's content filter
-                if hasattr(response.choices[0].message, "content"):
-                    label = response.choices[0].message.content.strip().replace("topic: ", "")
+                choice = response.choices[0]
+
+                if choice.finish_reason == "stop":
+                    label = choice.message.content.strip().replace("topic: ", "")
+                elif choice.finish_reason == "length":
+                    logger.warn(f"Extracing Topics - Length limit reached for doc_ids ({repr_doc_ids})")
+                    if hasattr(response.choices[0].message, "content"):
```

Comment:
I am not sure of the APIs behavior when hitting the token limit, so I kept the check for `content`.

Reply:
For now, this is fine I guess. Also, you can replace `response.choices[0]` with `choice`. Also, and this might be just me, could we replace the variable name `choice` with `output`? I find the term `choice` not easily readable/explicit.